crates/rpc/src/handlers/validator.rs

```diff
@@ -424,3 +430,61 @@ fn check_validator_participation(
         Ok(validator.is_active_validator(epoch))
     }
 }
+#[get("/validator/attestation_data")]
+pub async fn get_attestation_data(
+    db: Data<ReamDB>,
+    opertation_pool: Data<Arc<OperationPool>>,
+    query: Query<AttestationQuery>,
+) -> Result<impl Responder, ApiError> {
+    let store = Store {
+        db: db.get_ref().clone(),
+        operation_pool: opertation_pool.get_ref().clone(),
+    };
+
+    if store.is_syncing().map_err(|err| {
+        ApiError::InternalError(format!("Failed to check syncing status, err: {err:?}"))
+    })? {
+        return Err(ApiError::UnderSyncing);
+    }
+
+    let slot = query.slot;
+
+    let current_slot = store
+        .get_current_slot()
+        .map_err(|err| ApiError::InternalError(format!("Failed to slot_index, error: {err:?}")))?;
+
+    if slot > current_slot + 1u64 {
+        return Err(ApiError::InvalidParameter(format!(
+            "Slot {slot:?} is too far ahead of the current slot {current_slot:?}"
+        )));
+    }
```

Comment:
```suggestion
    if slot > current_slot + 1 {
        return Err(ApiError::InvalidParameter(format!(
            "Slot {slot:?} is too far ahead of the current slot {current_slot:?}"
        )));
    }
```

the type declaration here isn't needed